src/main/java/org/llorllale/cactoos/matchers/IsNumber.java

```diff
@@ -35,19 +35,32 @@
  */
 public final class IsNumber extends MatcherEnvelope<Number> {
 
+    /**
+     * Comparator of numbers.
+     */
+    private static final Comparator<Number> FNC =
+        Comparator
+            .comparing(Number::doubleValue)
+            .thenComparing(Number::intValue)
+            .thenComparing(Number::longValue)
+            .thenComparing(Number::floatValue);
+
     /**
      * Ctor.
      * @param expected The expected value
      */
     public IsNumber(final Number expected) {
         super(
             new MatcherOf<>(
-                expected,
-                Comparator
-                    .comparing(Number::doubleValue)
-                    .thenComparing(Number::intValue)
-                    .thenComparing(Number::longValue)
-                    .thenComparing(Number::floatValue)
+                actual -> IsNumber.FNC.compare(actual, expected) == 0,
+                desc -> desc.appendText("equals ").appendValue(expected),
+                (actual, desc) -> desc
+                    .appendText("comparator returns ")
+                    .appendValue(IsNumber.FNC.compare(actual, expected))
+                    .appendText(" when ")
+                    .appendValue(expected)
+                    .appendText(" compared to ")
+                    .appendValue(actual)
```

Comment:
@baudoliver7 let's add a todo to introduce a `ComparatorMatcher` that encapsulates this logic here: it would only take a `Comparator<X>` and an expected `X` for example.

Reply:
@victornoel Good